File: archivist/src/test/java/com/zorroa/archivist/repository/MigrationDaoTests.java
package com.zorroa.archivist.repository;

import com.zorroa.archivist.AbstractTest;
import com.zorroa.archivist.domain.Migration;
import com.zorroa.archivist.domain.MigrationType;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import static junit.framework.TestCase.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

/**
 * Created by chambers on 2/3/16.
 */
public class MigrationDaoTests extends AbstractTest {

    @Autowired
    MigrationDao migrationDao;

    @Test
    public void testGetAll() {
        int count = jdbc.queryForObject("SELECT COUNT(1) FROM migration", Integer.class);
        assertEquals(count, migrationDao.getAll().size());
    }

    @Test
    public void testGetAllByType() {
        int count = jdbc.queryForObject("SELECT COUNT(1) FROM migration WHERE int_type=?", Integer.class,
                MigrationType.ElasticSearchIndex.ordinal());
        assertEquals(count, migrationDao.getAll(MigrationType.ElasticSearchIndex).size());
    }

    @Test
    public void setVersion() {
        for (Migration m: migrationDao.getAll()) {
            assertTrue(migrationDao.setVersion(m, 1000000));
        }

        for (Migration m: migrationDao.getAll()) {
            assertFalse(migrationDao.setVersion(m, 1000000));
        }
    }
}

File: archivist/src/test/kotlin/com/zorroa/archivist/repository/MigrationDaoTests.kt
package com.zorroa.archivist.repository

import com.zorroa.archivist.AbstractTest
import com.zorroa.archivist.domain.Migration
import com.zorroa.archivist.domain.MigrationType
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired

import junit.framework.TestCase.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue

/**
 * Created by chambers on 2/3/16.
 */
class MigrationDaoTests : AbstractTest() {

    @Autowired
    internal var migrationDao: MigrationDao? = null

    @Test
    fun testGetAll() {
        val count = jdbc.queryForObject("SELECT COUNT(1) FROM migration", Int::class.java)!!
        assertEquals(count, migrationDao!!.getAll().size)
    }

    @Test
    fun testGetAllByType() {
        val count = jdbc.queryForObject("SELECT COUNT(1) FROM migration WHERE int_type=?", Int::class.java,
                MigrationType.ElasticSearchIndex.ordinal)
        assertEquals(count, migrationDao!!.getAll(MigrationType.ElasticSearchIndex).size)
    }

    @Test
    fun setVersion() {
        for (m in migrationDao!!.getAll()) {
            assertTrue(migrationDao!!.setVersion(m, 1000000))
        }

        for (m in migrationDao!!.getAll()) {
            assertFalse(migrationDao!!.setVersion(m, 1000000))
        }
    }
}
